platformio.ini

```diff
@@ -19,107 +19,37 @@ extra_configs =
   platformio_override.ini
 
 [common]
-# ------------------------------------------------------------------------------
-# PLATFORM:
-#   !! DO NOT confuse platformio's ESP8266 development platform with Arduino core for ESP8266
-#
-#   arduino core 2.6.3 = platformIO 2.3.2
-#   arduino core 2.7.0 = platformIO 2.5.0
-# ------------------------------------------------------------------------------
-arduino_core_2_6_3 = espressif8266@2.3.3
-arduino_core_2_7_4 = espressif8266@2.6.2
-arduino_core_3_0_0 = espressif8266@3.0.0
-arduino_core_3_0_2 = espressif8266@3.2.0
-arduino_core_3_1_0 = espressif8266@4.1.0
-arduino_core_3_1_2 = espressif8266@4.2.1
-
-# Development platforms
-arduino_core_develop = https://github.com/platformio/platform-espressif8266#develop
-arduino_core_git = https://github.com/platformio/platform-espressif8266#feature/stage
-
-# Platform to use for ESP8266
-platform_wled_default = ${common.arduino_core_3_1_2}
-# We use 2.7.4.7 for all, includes PWM flicker fix and Wstring optimization
-#platform_packages = tasmota/framework-arduinoespressif8266 @ 3.20704.7
-platform_packages = platformio/toolchain-xtensa @ ~2.100300.220621 #2.40802.200502
```

Comment:
@blazoncek @willmmiles I seems to remember that this was added on purpose, especially to prevent very strange problems observed by users making their own custom builds? I think we need the `platform_packages`  section for 8266?

Reply:
We'd removed an unversioned `platform_packages = platformio/framework-arduinoespressif8266` call because it was overriding the `platform = ${common.platform_wled_default}` selection.  The unversioned reference was picking up old random versions from people's build systems instead of the specific framework version intended by the `platform` entry.

I can't speak for specific the specific toolchain version selections - it's "always been that way" for the few months I've been working with WLED.